articles/101_build_tool.md

```diff
@@ -0,0 +1,213 @@
+For each package the documentation usually describes what the dependencies are, how to setup the environment to build the package as well as how to setup the environment afterwards to use the package.
+From an efficiency point of view that manual process does not scale well.
+
+This article describes the steps from the current build tools used in the ROS ecosystem to a single universal build tool which automates the process and works for various different use cases.
+
+## Goal
+
+The goal of the build tool is to build a set of packages with a single invocation automating the process.
+It should work with ROS 1 packages as well as ROS 2 packages which provide the necessary information in their manifest files.
+It should also work with packages that do not provide manifest files themselves, given that the meta information is externally provided.
+This will allow the build tool to be utilized for non-ROS dependencies of ROS packages (e.g. Gazebo including its dependencies).
+
+### Out of Scope
+
+The build tool does not cover the steps necessary to fetch the sources of the to-be-built packages.
+This can be done with existing tools (e.g. [rosinstall_generator](http://wiki.ros.org/rosinstall_generator), [vcstool](https://github.com/dirk-thomas/vcstool)).
+
+The build tool also does not provide a mechanism to install any dependencies required to build the packages.
+In the ROS ecosystem [rosdep](http://wiki.ros.org/rosdep) can be used for this.
+
+The build tool also does not create binary packages (e.g. a Debian package).
+In the ROS ecosystem [bloom](http://wiki.ros.org/bloom) is used to generate the required metadata and then platform dependent tools like `dpkg-buildpackage` build binary packages.
+
+## Build Tool vs. Build System
+
+A build tool operates on a set of packages.
+It determines the dependency graph and invokes the package specific build tool of each package in topological order.
+The build tool itself should know as little as possible about the build system used for a specific package.
+Just enough in order to know how to setup the environment for it, invoke the build, and setup the environment to use the built package.
+
+The build system on the other hand operates on a single package.
+In the case a package uses e.g. `CMake` the responsibility of the tool comes down to invoke the common steps `cmake`, `make`, `make install` for this package.
+As another example for a package using `Autotools` the steps could look like `configure`, `make`, `make install`.
+
+### Environment Setup
+
+A very important part beside the actual build of a package is the environment setup.
+For a CMake package e.g. a dependency discovered using the CMake function `find_package` needs to be either in a common location or the location must be provided through the environment variable `CMAKE_PREFIX_PATH`.
+On the other hand after a package has been built and installed the environment might need to be extended to use the package.
```

Comment:
I think this could be clearer. I know you mean "On the other hand" where the other hand is running something rather than building on top of it, but that's isn't clear enough I think. Perhaps:

> In addition to building a package on top of another package (using `find_package` in the case of CMake), you may need to adjust the environment in order to run an executable from a package.

Really just need to clarify what "use" means in "to use the package". So you can also just modify what you have to be clearer. This ambiguity isn't really addressed by the example that comes next in my opinion either. It could be cleared up by explaining the shared libraries may need to be on a particular PATH so they can be loaded at runtime of an executable.